fms2_io/netcdf_io.F90

```diff
@@ -45,6 +45,7 @@ module netcdf_io_mod
 integer, parameter, public :: max_num_compressed_dims = 10 !> Maximum number of compressed
                                                            !! dimensions allowed.
 integer, private :: fms2_ncchksz = -1 !< Chunksize (bytes) used in nc_open and nc_create
+integer, private :: fms2_header_buffer_val = 16384  !< value used in NF__ENDDEF
```

Comment:
You should initialize this to some error value.  It should always be set in `netcdf_io_init`.  See the `fms2_ncchksz` example.